pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,554 @@
+}
+
+func curlCommandforInternet(target string) []string {
+	return []string{"curl", "-s", "--fail", target}
+}
+
+func newService(name string, selector map[string]string, portName string, port int) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{Name: name, Port: int32(port)},
+			},
+			Selector: selector,
+		},
+	}
+}
+
+type deploymentParameters struct {
+	Name        string
+	Kind        string
+	Image       string
+	Replicas    int
+	Port        int
+	Command     []string
+	Affinity    *corev1.Affinity
+	Tolerations []corev1.Toleration
+	Labels      map[string]string
+}
+
+func newDeployment(p deploymentParameters) *appsv1.Deployment {
+	if p.Replicas == 0 {
+		p.Replicas = 1
+	}
+	replicas32 := int32(p.Replicas)
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: p.Name,
+			Labels: map[string]string{
+				"name": p.Name,
+				"kind": p.Kind,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: p.Name,
+					Labels: map[string]string{
+						"name": p.Name,
+						"kind": p.Kind,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: p.Name,
+							Env: []corev1.EnvVar{
+								{Name: "PORT", Value: fmt.Sprintf("%d", p.Port)},
+							},
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: int32(p.Port)},
+							},
+							Image:           p.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         p.Command,
+						},
+					},
+					Affinity: p.Affinity,
+				},
+			},
+			Replicas: &replicas32,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": p.Name,
+					"kind": p.Kind,
+				},
+			},
+		},
+	}
+}
+
+type k8sConnectivityParams struct {
+	clients         *deploymentClients
+	client          k8sClientOperations
+	params          Parameters
+	antreaNamespace string
+	clientPods      *corev1.PodList
+	echoPods        map[string]string
+}
+
+func NewAntreaConnectivityCheck(client k8sClientOperations, p Parameters) *k8sConnectivityParams {
+	return &k8sConnectivityParams{
+		client:          client,
+		antreaNamespace: "kube-system",
+		params:          p,
+	}
+}
+
+func (k *k8sConnectivityParams) validatePodToPod(ctx context.Context) {
+	//conducts Pod to Pod connectivity tests within same node and different node
+	for _, clientPod := range k.clientPods.Items {
+		for echoName, echoIP := range k.echoPods {
+			var (
+				srcPod  = connectivityCheckNamespace + "/" + clientPod.Name
+				dstPod  = connectivityCheckNamespace + "/" + echoName
+				success = true
+			)
+			k.Header("Validating from pod %s to pod %s...", srcPod, dstPod)
+			_, err := k.client.ExecInPod(ctx, connectivityCheckNamespace, clientPod.Name, "", curlCommand(echoIP+":80"))
```

Comment:
I don't understand. I don't think `curl` is available in the `agnhost` container image?

Reply:
thanks for this, I mistakenly used my local environment inbuilt curl command for testing.
I've changed it to /agnhost connect now